packages/deploy-script-support/README.md

```diff
@@ -15,93 +16,105 @@ it uses, identifies the "manifest", (which associates permissions to access
+available to the proposal as `valueVowRef` in options.
+
+The manifest gives the permissions for accessing the promise space, and passes
+the `installations` to the proposalBuilder. Notice that `valueVowRef` from the
+proposalBuilder is passed to `getManifest`, which adds it to `installations`,
+with the name `valueVow`. The latter must match the well-known name in
+`consume.installation` and `agoricNames.installation`. The call to `restoreRef`
+causes the installation to be registered there.
+
+```
+export const getManifestForValueVow = ({ restoreRef }, { valueVowRef }) => {
+  console.log('valueVowRef', valueVowRef);
+  return {
+    manifest: {
+      [startValueVow.name]: {
+        consume: {
+          startUpgradable: true,
+        },
+        installation: {
+          consume: { valueVow: true },
+        },
+        instance: {
+          produce: { valueVow: true },
+        },
+      },
+    },
+    installations: {
+      valueVow: restoreRef(valueVowRef),
+    },
+  };
+};
+```
 
 ### Proposal
 
 The proposal is called with `(powers, options)` available. The manifest
 detailing the powers that will be used is usually in the same file, and
 conventionally provided by a function named `getManifestForFoo`. The manifest
 needs to have a unique name, since it will be referenced by name from the
-script. The usual format is
-```js
-export const foo = async (
-  {
-    consume: {
-      ...
-    },
-    brands: {
-      ...
-    }
-  },
- options,
-) => {
-  const { fooRef } = options;
-  // do the things using powers and options
-};
-
-export const getManifestForFoo = (powers, options) => {
-  manifest: {
-     [foo.name]: {
-       consume: {
-         ...
-      },
-  options,
-)};
-```
+script. 
 
 `manifest` contains descriptions of powers to be provided to the proposals.
 
-**TODO**  what happens with the `installations` in [`startPsm.js`](https://github.com/Agoric/agoric-sdk/blob/b13743a2cccf0cb63a412b54384435596d4e81ea/packages/inter-protocol/src/proposals/startPSM.js#L496)?
+`options` can provide the proposal with arbitrary other powerful objects.
 
-`options` allows the proposal to be provided with arbitray other powerful
-objects.
+`installations` are  passed to the proposal,and are useful for creating
+instances of contracts with Zoe. When upgrading contracts, Zoe requires
+bundleIds rather than installations. The bundleId can be retrieved from the
+`contractRef` from the proposalBuilder, so that is usually passed through in
+options.
 
 ### proposalBuilder Script
 
-The script describes how to build the core proposal. For
-`agoric-3-proposals` and uploading to the chain, the script must be named in the
-`CoreProposalSteps` section in [`upgrade.go`](../../golang/cosmos/app/upgrade.go),
-and its `defaultProposalBuilder` will be invoked directly.
+The script describes how to build the core proposal. The process is different
+for SoftwareUpgrades and coreEvals. Script files should export
```

Comment:
hm. different how? I didn't realize that there are different kinds of `CoreEvalBuilder`.

Reply:
> SoftwareUpgrades and coreEvals

one is capitalized but the other is not?

`CoreEvalProposal` and `SoftwareUpgradeProposal` are the in-protocol protobuf message names, FWIW.